m2cgen/assemblers/boosting.py

```diff
@@ -194,6 +195,78 @@ def _count_leaves(tree):
         return num_leaves
 
 
+class XGBoostLinearModelAssembler(ModelAssembler):
+
+    classifier_name = "XGBClassifier"
+
+    def __init__(self, model):
+        super().__init__(model)
+
+        self._output_size = 1
+        self._is_classification = False
+
+        model_class_name = type(model).__name__
+        if model_class_name == self.classifier_name:
+            self._is_classification = True
+            if model.n_classes_ > 2:
+                self._output_size = model.n_classes_
+
+        self._base_score = model.base_score
+
+        model_dump = model.get_booster().get_dump(dump_format="json")
+        self.weight = json.loads(model_dump[0])["weight"]
+        self.bias = json.loads(model_dump[0])["bias"]
+
+    def assemble(self):
+        if self._is_classification:
+            if self._output_size == 1:
+                self.bias = self.bias[0]
+                return self._assemble_bin_class_output(self.weight, self.bias)
+            else:
+                return self._assemble_multi_class_output(
+                    self.weight, self.bias)
+        else:
+            self.bias = self.bias[0]
+            return self._assemble_single_output(self.weight, self.bias)
+
+    def _assemble_single_output(self, weights, bias):
+        coef = utils.to_1d_array(weights)
+
+        expr = _linear_to_ast(coef, bias)
+
+        result_ast = utils.apply_bin_op(
+            ast.NumVal(self._base_score),
+            expr,
+            ast.BinNumOpType.ADD)
+
+        return ast.SubroutineExpr(result_ast)
+
+    def _assemble_bin_class_output(self, weights, bias):
+        # Base score is calculated based on https://github.com/dmlc/xgboost/blob/master/src/objective/regression_loss.h#L64  # noqa
+        # return -logf(1.0f / base_score - 1.0f);
+        if self._base_score != 0:
```

Comment:
Not a big fan of mutating a state during the operation which should be idempotent. Basically doing means that the same assembler instance can't be reused for different models, which is very far from obvious.